### PR TITLE
feat: make the controller and injector log levels configurable

### PR DESCRIPTION
**Pull request by afrancis101**
## What does this PR do?

- [x] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- `Allows the log level of the controller and injector to be configurable. Currently the log level is set to debug within the code and cannot be changed at runtime`

## Code Quality Checklist

- [x] The documentation is up to date.
- [x] My code is sufficiently commented and passes continuous integration checks.
- [x] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [x] I leveraged continuous integration testing
    - [x] by depending on existing `unit` tests or `end-to-end` tests.
    - [x] by adding new `unit` tests or `end-to-end` tests.
- [x] I manually tested the following steps:
    - `x`
    - [x] locally.
    - [x] as a canary deployment to a cluster.

### CODE REVIEW

File: log/log.go [hunk 1]
```diff
@@ -18,6 +20,17 @@ func NewZapLogger() (*zap.SugaredLogger, error) {
 	loggerConfig.EncoderConfig.MessageKey = "message"
 	loggerConfig.EncoderConfig.EncodeTime = zapcore.EpochMillisTimeEncoder
 
+	// optionally override the log level from the default based on the LOG_LEVEL env var
+	lvl, exists := os.LookupEnv("LOG_LEVEL")
```
Comment (ptnapoleon) [inline log/log.go:24]: @clairecng, I'm fairly fine with controlling this via a `LOG_LEVEL` env var, as that's a typical practice, but do we also want to pass this in via config, the way we are with injector log level?

File: config/config.go [hunk 2]
```diff
@@ -291,6 +292,12 @@ func New(logger *zap.SugaredLogger, osArgs []string) (config, error) {
 		return cfg, err
 	}
 
+	mainFS.StringVar(&cfg.Injector.LogLevel, "injector-log-level", "", "The LOG_LEVEL used for the injector pods")
```
Comment (ptnapoleon) [inline config/config.go:295]:
```suggestion
	mainFS.StringVar(&cfg.Injector.LogLevel, "injector-log-level", "DEBUG", "The LOG_LEVEL used for the injector pods")
```

Could we have a default when unspecified?

Comment (afrancis101) [reply config/config.go]: Sure, I've made the change.

**Comment by ptnapoleon** [comment]
Could you please sign your commit?

**Comment by afrancis101** [comment]
> Could you please sign your commit?

My bad. The commit has been signed.

**Pull request merged**